kuma/javascript/src/payments/subscription-form.jsx

```diff
@@ -0,0 +1,107 @@
+// @flow
+import * as React from 'react';
+import { useRef, useState } from 'react';
+
+import { getLocale, gettext, Interpolated } from '../l10n.js';
+
+export default function SubscriptionForm() {
+    const locale = getLocale();
+    const subscriptionFormRef = useRef(null);
+    const [subscribeFormDisabled, setSubscribeFormDisabled] = useState(false);
+    const [subscribeButtonEnabled, setSubscribeButtonEnabled] = useState(false);
+
+    const toggleButton = () => {
+        setSubscribeButtonEnabled(!subscribeButtonEnabled);
+    };
+
+    /**
+     * Opens Stripe modal allowing a user to complete their subscription.
+     * @param {Object} event - The form submit event
+     */
+    const submit = event => {
+        event.preventDefault();
+
+        const subscriptionForm = subscriptionFormRef.current;
+        if (subscriptionForm) {
+            const formData = new FormData(subscriptionForm);
+            setSubscribeFormDisabled(true);
+
+            const stripeHandler = window.StripeCheckout.configure({
+                key: window.mdn.stripePublicKey,
+                locale: 'auto',
+                name: 'MDN Web Docs',
+                zipCode: true,
+                currency: 'usd',
+                amount: 500,
+                email: '',
+                token: function(response) {
+                    formData.set('stripe_token', response.id);
+                    subscriptionForm.submit();
+                },
+                closed: function() {
+                    if (!formData.get('stripe_token')) {
+                        setSubscribeFormDisabled(false);
+                    }
+                }
+            });
+
+            stripeHandler.open();
+        }
+    };
+
+    return (
+        <div className="subscriptions-form">
+            <header className="subscriptions-form-header">
+                <h2>
+                    <Interpolated
+                        id={gettext('$5 <perMontSub />')}
+                        perMontSub={<sub>{gettext('/mo')}</sub>}
+                    />
+                </h2>
+            </header>
+            <form
+                ref={subscriptionFormRef}
+                name="subscription-form"
+                method="post"
+                onSubmit={submit}
+                disabled={subscribeFormDisabled}
+            >
+                <label className="payment-opt-in">
+                    <input
+                        type="checkbox"
+                        required="required"
+                        onClick={toggleButton}
```

Comment:
We should go full [controlled component](https://reactjs.org/docs/forms.html#controlled-components), i.e. also set the `<input>`'s `value=` based on `subscribeButtonEnabled`. It is also behaving opposite of what is expected for me, i.e. when checking the box the button is disabled.

Also my renaming advice was a little shortsighted for that one. I'd go with something more meaningful in this context like: `paymentAuthorized`.
Similar nit/advice for the other flag: I'd go with `isSubmitting`.
The upside of these renames is that the state now reflects user intent and not what UI might result from it. Now when we want to make other UI changes based on it, we don't have to rename it.

Reply:
I had a look at the `checkbox` in order to set a value on it but noticed that this field in the current implementation does not have a `name` attribute:

```
<input type="checkbox" required="required">
```

This means that the data of the field is not sent to the server, no? Looking at the Python for `create_stripe_subscription` it also does not look like the backend is checking for the field or a specific value.

I am guessing we want to change this?

Perhaps something like:

```
<input type="checkbox" name="payment_authorized" required="required">
```

With that, do we want to do validation on the backend if the box was not checked? I mean, technically, someone should not be able to submit the form if they did not check the checkbox.

We also are not storing this anywhere so, should we be?